### PR TITLE
SDK image unit tests

### LINKED ISSUE #86: Create more extensive automated tests for SDK images
Some of the secondary components that are included in the build images are not covered by the automated tests that run as part of CI.  Examples include VS Test Agent, Nuget CLI, PATH ENV contains the expected paths, etc.

### PR DESCRIPTION
**Pull request by mthalman**
Adds some more unit tests for the SDK images:
* Verify Web Deploy is installed
* Verify ClickOnce publishing.  This uses a 3.5 and 4.8 project that is configured with the necessary ClickOnce publishing properties.  The Dockerfile just publishes the project as a ClickOnce app to an output folder which the test verifies the content of.
* Verifies default shell is set to PowerShell.  To avoid the overhead of additional test artifacts and test execution time, this is verified by simply adding a simple PowerShell command in the Dockerfiles of the 3.5/4.8 test console projects.  Docker build will fail if shell isn't set correctly in that case.

Fixes #86

### CODE REVIEW

File: tests/Microsoft.DotNet.Framework.Docker.Tests/projects/dotnetapp-3.5/Dockerfile [hunk 1]
```diff
@@ -3,6 +3,9 @@ ARG BASE_RUNTIME_IMAGE
 
 
 FROM $BASE_BUILD_IMAGE as builder
```
Comment (MichaelSimons) [inline tests/Microsoft.DotNet.Framework.Docker.Tests/projects/dotnetapp-3.5/Dockerfile]: I realize this issue is SDK specific, but I think verifying the default SHELL more holistically would be beneficial.  We've had issues in this space in the past.  I think you could easily verify this with a shared test that executes something similar to `docker inspect mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-1909 -f "{{ .Config.Shell }}"`  I expect all images except the runtime should return the same `powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';` value.

File: tests/Microsoft.DotNet.Framework.Docker.Tests/SdkOnlyImageTests.cs [hunk 2]
```diff
@@ -152,5 +152,47 @@ public void VerifyGacUtil(ImageDescriptor imageDescriptor)
...
+
+        [SkippableTheory("4.6.2", "4.7", "4.7.1", "4.7.2")]
+        [Trait("Category", "sdk")]
+        [MemberData(nameof(GetImageData))]
+        public void VerifyWebDeploy(ImageDescriptor imageDescriptor)
+        {
+            string baseBuildImage = ImageTestHelper.GetImage("sdk", imageDescriptor.Version, imageDescriptor.OsVariant);
+            string appId = $"webdeploy-{DateTime.Now.ToFileTime()}";
+            
+            // Calling msdeploy's help command results in a -1 exit code. To prevent that from bubbling up and causing the
```
[the remark this answers is not in the document]
Comment (mthalman) [reply tests/Microsoft.DotNet.Framework.Docker.Tests/SdkOnlyImageTests.cs]: I've logged it: https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1047413 (internal MSFT link)

File: tests/Microsoft.DotNet.Framework.Docker.Tests/projects/dotnetapp-3.5/Dockerfile (hunk 1 above)
Comment (mthalman) [reply tests/Microsoft.DotNet.Framework.Docker.Tests/projects/dotnetapp-3.5/Dockerfile]: I like that approach.

File: tests/Microsoft.DotNet.Framework.Docker.Tests/DockerHelper.cs [hunk 3]
```diff
@@ -191,6 +191,10 @@ private string ExecuteWithLogging(string args, bool ignoreErrors = false, bool a
             ExecuteWithLogging(
                 $"inspect -f \"{{{{(index (index .NetworkSettings.Ports \\\"{containerPort}/tcp\\\") 0).HostPort}}}}\" {container}");
 
+        public string GetImageShell(string imageTag) =>
+            ExecuteWithLogging(
+                "inspect -f \"{{ .Config.Shell }}\" " + imageTag);
```
Comment (MichaelSimons) [inline tests/Microsoft.DotNet.Framework.Docker.Tests/DockerHelper.cs]: Consider utilizing string interpolation.
```suggestion
                $"inspect -f \"{{{{ .Config.Shell }}}}\" {imageTag}");
```

File: tests/Microsoft.DotNet.Framework.Docker.Tests/projects/dotnetapp-3.5/Dockerfile [hunk 4]
```diff
@@ -3,6 +3,7 @@ ARG BASE_RUNTIME_IMAGE
...
+
```
Comment (MichaelSimons) [inline tests/Microsoft.DotNet.Framework.Docker.Tests/projects/dotnetapp-3.5/Dockerfile]: Consider reverting to keep the format of the test Dockerfiles consistent across versions.

File: tests/Microsoft.DotNet.Framework.Docker.Tests/projects/dotnetapp-4.8/Dockerfile [hunk 5]
```diff
@@ -3,6 +3,7 @@ ARG BASE_RUNTIME_IMAGE
 
 
 FROM $BASE_BUILD_IMAGE as builder
+
```
Comment (MichaelSimons) [inline tests/Microsoft.DotNet.Framework.Docker.Tests/projects/dotnetapp-4.8/Dockerfile]: Consider reverting to keep the format of the test Dockerfiles consistent across versions.

**Pull request merged**